lib/telemetry/armon_metrics.go

```diff
@@ -327,57 +401,92 @@ func circonusSink(cfg TelemetryConfig, hostname string) (metrics.MetricSink, err
 	return sink, nil
 }
 
-// InitTelemetry configures go-metrics based on map of telemetry config
-// values as returned by Runtimecfg.Config().
-func InitTelemetry(cfg TelemetryConfig) (*metrics.InmemSink, error) {
+// addSink takes our collection of sinks and applies configuration to its setup function, appending the new sink to the
+// collection.
+func addSink(sinks metrics.FanoutSink, fn sinkFn, cfg Config, hostname string) error {
+	// Build the sink
+	s, err := fn(cfg, hostname)
+	if err != nil {
+		return err
+	}
+	// Compose into FanoutSink
+	if s != nil {
+		sinks = append(sinks, s)
+	}
+	return nil
+}
+
+// initSinks composes all of our sink options from RuntimeCfg into a FanoutSink
+func initSinks(cfg Config, hostname string) (metrics.FanoutSink, error) {
+	var sinks metrics.FanoutSink
+	// Compose all of our external sinks with FanoutSink.
+	// All sink inits must succeed - we abort setup if any of the configuration is invalid.
+	if err := addSink(sinks, statsiteSink, cfg, hostname); err != nil {
```

Comment:
I believe this `hostname` parameter could be removed. It is only used by the datadog sink, but looking at the code for that the datadog sink only uses it if `EnableHostNamePropagation` is called, and I don't see that being called anywhere, so I think the value is ignored in that case as well.

I think this is not by accident, because `go-metrics` handles adding this label for all sinks, so we don't need it pass it to the sinks themselves.

Reply:
Yeah, that param looked a bit fishy when I was going through this. I've removed it in the sinkFns